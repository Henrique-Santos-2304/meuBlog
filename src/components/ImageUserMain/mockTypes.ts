export const mockTypes = {
  data: {
    alt: "photo",
    ariaLabel: "uma img qualquer",
    imageUser: {
      name: "photo",
      url: "img.imagen.png",
    },
  },
  links: {
    name: "Henrique dos Santos",
    profission: "Desenvolvedor FrontEnd",
    platform: "Web, Mobile, Desktop",
    linksSocials: [],
    email: {
      title: "icons",
      email: "henrique.multitech@gmail.com",
      ariaLabel: "Envio de email via site",
      id: "name",
    },
  },
};
