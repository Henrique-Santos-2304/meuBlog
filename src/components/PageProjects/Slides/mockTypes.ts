export const mockTypes = {
  cardsSlides: [
    {
      miniDescription: "Uma Descrição dos Slides",
      title: "Ver projetos",
      __typename: "nada",
      buttons: [
        { url: "ir.com", label: "fui" },
        { url: "ir.com2", label: "fui2" },
      ],
      imageFront: {
        alt: "descrição da imagem",
        ariaLabel: "essa imagem",
        imageUser: {
          url: "img/img.png",
        },
      },
    },
  ],
};
