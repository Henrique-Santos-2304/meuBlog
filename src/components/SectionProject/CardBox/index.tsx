import * as S from "./styles";
import Link from "next/link";

type cardsSlides = {
  title: string;
  img: string;
  buttonCode: string;
  marginBox: string;
};
const CardBox = ({ title, img, buttonCode, marginBox }: cardsSlides) => (
  <S.ContentSection marginBox={marginBox} aria-label="Titulo do projeto">
    <S.ContentTitle>
      <S.TitleProject>{title}</S.TitleProject>
    </S.ContentTitle>
    <S.Wrapper aria-label="Imagem do projeto">
      <S.PhotoProject src={img} layout="fill" quality={100} />
    </S.Wrapper>
    <S.ContentButton>
      <Link href={buttonCode} passHref>
        <a target="_blank">
          <S.ButtonCode aria-label="Link para código do Projeto">
            <S.Icon />
            Ver Código
          </S.ButtonCode>
        </a>
      </Link>
    </S.ContentButton>
  </S.ContentSection>
);

export default CardBox;
