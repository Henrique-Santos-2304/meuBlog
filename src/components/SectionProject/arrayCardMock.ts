export const images = [
  {
    title: "LeScone",
    img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634678383/Lescone_9fb07ad59b.gif",
    buttonCode: "https://github.com/Henrique-Santos-2304/Restaurante-LeScone",
  },
  {
    title: "Multitech",
    img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634678422/Multitech_desktop_90fabbf476_83baa085c2.gif",
    buttonCode: "https://github.com/Henrique-Santos-2304/Multitech",
  },
  {
    title: "HdcAgency",
    img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634845622/hdc_Agency_f400e88579.gif",
    buttonCode:
      "https://github.com/Henrique-Santos-2304/Agencia-de-Desenvolvedores",
  },
];
