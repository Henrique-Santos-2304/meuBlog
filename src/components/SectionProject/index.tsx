import * as S from "./styles";

const SectionProject = () => (
  <S.Wrapper id="project">
    <h1>SectionProject</h1>
  </S.Wrapper>
);

export default SectionProject;
