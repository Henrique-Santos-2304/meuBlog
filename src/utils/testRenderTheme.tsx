import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
