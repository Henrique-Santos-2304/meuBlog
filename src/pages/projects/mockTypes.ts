import { mockTypes } from "components/PageProjects/Slides/mockTypes";

export const mockTypesProject = {
  title: "Meus Projetos",
  cardsSlides: [...mockTypes.cardsSlides],
  metaDescription: {
    __typename: "",
    titlePage: "Projetos de Henrique",
    iconPage: [{ url: "imag.png", __typename: "" }],
  },
};
