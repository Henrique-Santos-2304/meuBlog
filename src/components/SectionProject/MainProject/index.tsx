/* eslint-disable @typescript-eslint/no-explicit-any */
import * as S from "./styles";

import Headings from "components/globalComponents/Headings";
import { images } from "../arrayCardMock";
import SliderPhoto from "components/globalComponents/SliderPhoto";

const SectionProject = () => {
  return (
    <S.Wrapper id="projects">
      <S.Title>
        <Headings
          msg="Meus Projetos"
          fontSizeText="2rem"
          marginText="1.5rem 0 1.4rem 3rem"
          aria-label="Titulo da seção"
        />
      </S.Title>
      <S.ContentSlide>
        <SliderPhoto
          images={images}
          receptorCards="projects"
          slidesToShow={1}
          marginBox="2rem 0 1rem"
        />
      </S.ContentSlide>
    </S.Wrapper>
  );
};

export default SectionProject;
