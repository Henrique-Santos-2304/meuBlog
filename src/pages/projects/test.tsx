import "match-media-mock";
import { screen } from "@testing-library/react";
import SectionProject from ".";
import { renderTheme } from "utils/testRenderTheme";
import { mockTypesProject } from "./mockTypes";

describe("<SectionProject />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<SectionProject {...mockTypesProject} />);
    expect(container).toMatchSnapshot();
  });
  it("should have title of the page  ", () => {
    renderTheme(<SectionProject {...mockTypesProject} />);
    expect(
      screen.getByRole("heading", {
        name: /meus projetos/i,
      })
    ).toBeInTheDocument();
  });
});
