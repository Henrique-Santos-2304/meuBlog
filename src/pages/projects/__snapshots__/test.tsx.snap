// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionProject /> should render  1`] = `
<div>
   
  <div
    class="chakra-container css-1n49xi7"
  >
    <div
      class="css-gmuwbf"
    />
  </div>
</div>
`;
